### DISCUSSION TITLE
How to set axis_rule?

### DISCUSSION THREAD

**Discussion by hewg2008** [deepmodeling/deepmd-kit#38] [Q&A]
In your guide，you provide the following description:
 "axis_rule":	[0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0],

    "_comment":	" default rule: []",
    "_comment":	" user defined rule: for each type provides two axes, ",
    "_comment":	"                    for each axis: (a_or_r, type, idx)",
    "_comment":	"                    if type < 0, exclude type -(type+1)",
    "_comment":        "                    for water (O:0, H:1) it can be",
    "_comment":        "                    [0, 1, 0, 0, 1, 1, 0, 0, 0, 0, 1, 0]",

My system is CH(C:0, H:1), and I made exactly the same setting as your example, when I ran train.py, it came out with below traceback:

Traceback (most recent call last):
  File "/opt/deepmd/bin/dp_train", line 128, in <module>
    _main()
  File "/opt/deepmd/bin/dp_train", line 98, in _main
    _do_work(jdata, run_opt)
  File "/opt/deepmd/bin/dp_train", line 113, in _do_work
    data = DataSystem(systems, set_pfx, batch_size, test_size, rcut, run_opt)
  File "/opt/deepmd/bin/../lib/deepmd/DataSystem.py", line 31, in __init__
    self.data_systems.append(DataSets(ii, set_prefix))
  File "/opt/deepmd/lib/deepmd/Data.py", line 30, in __init__
    self.load_batch_set (self.train_dirs[self.set_count % self.get_numb_set()])
  File "/opt/deepmd/lib/deepmd/Data.py", line 166, in load_batch_set
    self.atom_ener_batch = self.atom_ener_batch[:, self.idx_map]
IndexError: index 1 is out of bounds for axis 1 with size 1.

Please kindly help. 

Thanks.

**Comment by jameswind** [comment]
Does your system only contain two atoms? One C and one H?

**Comment by hewg2008** [comment]
Yes, only two atom types with 24 atoms, one type is C, the other type is H.

**Comment by jameswind** [comment]
Then you cannot use the non-smooth version, which requires at least 3 atoms
to define the local frame. Please try the smooth one.

**Comment by hewg2008** [comment]
Thanks, but a little confused, my system has actually 24 atoms, much more than 3 atoms. Or do you mean the system should be with at least 3 atom types?

**Comment by hewg2008** [comment]
Already worked. Thanks for the kind help.
